src/main/scala/sample/stream/GroupLogFile.scala

```diff
@@ -25,21 +25,25 @@ object GroupLogFile {
     // read lines from a log file
     val logFile = new File("src/main/resources/logfile.txt")
 
-    import akka.stream.io.Implicits._ // add file sources to Source object or use explicitly: SynchronousFileSource(f)
-    Source.synchronousFile(logFile).
+    FileIO.fromFile(logFile).
       // parse chunks of bytes into lines
       via(Framing.delimiter(ByteString(System.lineSeparator), maximumFrameLength = 512, allowTruncation = true)).
       map(_.utf8String).
+      map {
+        case line@LoglevelPattern(level) => (level, line)
+        case line@other => ("OTHER", line)
+      }.
       // group them by log level
-      groupBy {
-        case LoglevelPattern(level) => level
-        case other                  => "OTHER"
+      groupBy(5, _._1).
+      fold(("", List.empty[String])) {
+        case ((_, list), (level, line)) => (level, line :: list)
```

Comment:
This cannot be fixed currently: what would be needed is `.prefixAndTail(0)` and we do _not_ want to show that to people in a tutorial.